flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SchedulerBase.java

```diff
@@ -280,23 +281,8 @@ private ExecutionGraph createExecutionGraph(
 			failoverStrategy);
 	}
 
-	/**
-	 * @deprecated Direct access to the execution graph by scheduler implementations is discouraged
-	 * because currently the execution graph has various features and responsibilities that a
-	 * scheduler should not be concerned about. The following specialized abstractions to the
-	 * execution graph and accessors should be preferred over direct access:
-	 * <ul>
-	 *     <li>{@link #getSchedulingTopology()}
-	 *     <li>{@link #getFailoverTopology()}
-	 *     <li>{@link #getInputsLocationsRetriever()}
-	 *     <li>{@link #getExecutionVertex(ExecutionVertexID)}
-	 *     <li>{@link #getExecutionVertexId(ExecutionAttemptID)}
-	 *     <li>{@link #getExecutionVertexIdOrThrow(ExecutionAttemptID)}
-	 * </ul>
-	 * Currently, only {@link LegacyScheduler} requires direct access to the execution graph.
-	 */
-	@Deprecated
-	protected ExecutionGraph getExecutionGraph() {
+	@VisibleForTesting
+	public ExecutionGraph getExecutionGraph() {
```

Comment:
It will be needed by many tests to query the execution status when they are reworked to based on the new scheduler.
It's a prerequisite change for a set of following PRs so I think we can avoid adding one more hotfix to make it public then.

Reply:
Can you tell me which test(s) you are planning to port next?

Reply:
Regardless, I don't think we should remove the comment entirely. There is  a reason why helper methods such as `getExecutionVertex()` were introduced. There is business logic inside the `ExecutionGraph` and I would not want anyone to call it directly from the scheduler. We can add `@VisibleForTesting` but for me it does not necessarily mean that calling the method from production code is discouraged.

Reply:
> It's a prerequisite change for a set of following PRs so I think we can avoid adding one more hotfix to make it public then. 

If you think it makes your life easier, then we can leave this commit. However, I would prefer to change the visibility in non-hotfix commit when we rewrite the first test to use the new scheduler. In the git history it will be easier to see the justification. Moreover, if we delay rewritig the tests, the visibility will not be unnecessarily increased.

Reply:
There would be quite a few tests to be reworked. 
- ExecutionGraphDeploymentTest
- ExecutionGraphPartitionReleaseTest
- ExecutionGraphVariousFailuesTest
- ExecutionTest
- ExecutionVertexCancelTest
- ExecutionVertexInputConstraintTest
- ExecutionVertexTest
- ExecutionGraphNotEnoughResourceTest
- ExecutionGraphCheckpointCoordinatorTest
- ExecutionGraphColocationRestartTest
- ExecutionGraphSuspendTest
- FinalizeOnMasterTest

Most of them does not need much effort to be based on the new scheduler. Exposing ExecutionGraph would make the work much easier since we only need to force scheduling related actions to be conducted via the scheduler.

Reply:
I will think again about whether we must expose `getExecutionGraph()` to the tests.
Will drop this commit and may do it when reworking the tests.